mlir/lib/Quantum/Transforms/ChainedHadamard.cpp

```diff
@@ -40,15 +40,15 @@ struct ChainedHadamardOpRewritePattern : public mlir::OpRewritePattern<CustomOp>
             return failure();
 
         ValueRange qbs = op.getInQubits();
-        CustomOp parentHadamard = qbs[0].getDefiningOp<CustomOp>();
+        auto parentHadamard = dyn_cast<CustomOp>(qbs[0].getDefiningOp());
```

Comment:
Do you know what would happen if the `qbs` value was a block argument instead of being defined by an operation?

Reply:
Looking at the wording of the spec for `getDefiningOp`: 
```
If this value is the result of an operation, return the operation that defines it.
```
I think it would simply return a nullptr. But it is a useful experiment to try out!

Reply:
I would say adding a test might be worthwhile to make sure we don't error out :)

Reply:
Sounds good, I'll add it now